core/embed/rust/src/zcash_primitives/pallas/point.rs

```diff
@@ -0,0 +1,144 @@
+use core::ops::Deref;
+
+use pasta_curves::{
+    arithmetic::{CurveAffine, CurveExt},
+    group::{Curve, Group, GroupEncoding},
+    pallas::Point,
+    Fp,
+};
+
+use super::common::value_error;
+use crate::{
+    error::Error,
+    micropython::{
+        buffer::{Buffer, StrBuffer},
+        ffi,
+        gc::Gc,
+        map::Map,
+        obj::Obj,
+        qstr::Qstr,
+        typ::Type,
+        util,
+        wrap::{Wrapable, Wrapped},
+    },
+};
+
+pub static POINT_TYPE: Type = obj_type! {
+    name: Qstr::MP_QSTR_Point,
+    locals: &obj_dict!(obj_map! {
+            Qstr::MP_QSTR_extract => obj_fn_1!(point_extract).as_obj(),
+            Qstr::MP_QSTR_to_bytes => obj_fn_1!(point_to_bytes).as_obj(),
+            Qstr::MP_QSTR_is_identity => obj_fn_1!(point_is_identity).as_obj(),
+    }),
+    make_new_fn: point_from_bytes,
+    unary_op_fn: point_unary_op,
+    binary_op_fn: point_binary_op,
+};
+
+impl Wrapable for Point {
+    fn obj_type() -> &'static Type {
+        &POINT_TYPE
+    }
+}
+
+unsafe extern "C" fn point_from_bytes(
+    _type_: *const Type,
+    n_args: usize,
+    n_kw: usize,
+    args: *const Obj,
+) -> Obj {
+    let block = |args: &[Obj], _kwargs: &Map| {
+        if args.len() != 1 {
+            return Err(Error::TypeError);
+        }
+        let bytes: [u8; 32] = args[0].try_into()?;
+        let point = Point::from_bytes(&bytes);
+        match Option::<Point>::from(point) {
+            Some(p) => p.wrap(),
+            None => Err(value_error(b"Conversion failed.\0")),
+        }
+    };
+    unsafe { util::try_with_args_and_kwargs_inline(n_args, n_kw, args, block) }
+}
+
+unsafe extern "C" fn point_unary_op(op: ffi::mp_binary_op_t, this: Obj) -> Obj {
```

Comment:
i'm surprised that this compiles, given that the argument is supposed to be `mp_unary_op_t` ?

Reply:
```rust
pub type mp_unary_op_t = cty::c_uint;
pub type mp_binary_op_t = cty::c_uint;
```

Reply:
d86c1ae6130eb034a2773a958f847539b6c600d6